pkg/resources/opnicluster/services.go

```diff
@@ -678,6 +741,19 @@ func (r *Reconciler) drainDeployment() (runtime.Object, reconciler.DesiredState,
 	return deployment, deploymentState(r.spec.Services.Drain.Enabled), nil
 }
 
+func (r *Reconciler) trainingControllerDeployment() (runtime.Object, reconciler.DesiredState, error) {
+	deployment := r.genericDeployment(v1beta2.ServiceKind(v1beta1.TrainingControllerService))
+
+	deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env,
```

Comment:
this seems dangerous to add as a container environment variable

Reply:
So, the NODE_TLS_REJECT_UNAUTHORIZED environment variable is used for the Elasticdump command utilized by this service since it won't be working otherwise.

Reply:
If that is the case, it should be set only in the specific place it's used, not for the container.